app/controllers/previousRegistrations/PreviousEuCountryController.scala

```diff
@@ -63,10 +64,28 @@ class PreviousEuCountryController @Inject()(
           BadRequest(view(formWithErrors, waypoints, index)).toFuture,
 
         value =>
+          val existingCountry = request.userAnswers.get(PreviousEuCountryPage(index))
+
+          val cleanedAnswersTry: Try[UserAnswers] =
+            (existingCountry, Some(value)) match {
+              case (Some(oldCountry), Some(newCountry)) if oldCountry != newCountry =>
+                cleanUp(index, request.userAnswers)
+              case _ =>
+                Success(request.userAnswers)
+            }
           for {
-            updatedAnswers <- Future.fromTry(request.userAnswers.set(PreviousEuCountryPage(index), value))
+            cleanedAnswers <- Future.fromTry(cleanedAnswersTry)
+            updatedAnswers <- Future.fromTry(cleanedAnswers.set(PreviousEuCountryPage(index), value))
             _              <- cc.sessionRepository.set(updatedAnswers)
           } yield Redirect(PreviousEuCountryPage(index).navigate(waypoints, request.userAnswers, updatedAnswers).route)
       )
   }
+
+  private def cleanUp(index: Index, answers: UserAnswers): Try[UserAnswers] = {
```

Comment:
There's a cleanup method you can override on `PreviousEuCountryPage` to do this functionality, without having to call it, as soon as you update the answer, you can cleanup the other answers with the same index. You could also add a unit test for this